tools/src/main/scala/scala/scalanative/linker/Reach.scala

```diff
@@ -44,9 +44,22 @@ class Reach(
     // in reachUnavailable
     defns ++= done.valuesIterator.filter(_ != null)
 
+    val resultEntries = entries.head match {
```

Comment:
It seems quite dangerous to rely on `entries.head`. I wouldn't expect there to be any guarantee about the order or content of these entries.

Reply:
That's true, I've prepared a more generic solution that updates the original main with resolved one.
 Unfortunately, because of that, I needed to update all ReachabilitySuites - all of them were using `def main(): Unit` instead of `def main(_: Array[String]): Unit` which is not a real main method (sbt would not detect it as an entry point)